File: backend/app/models/scammer_models.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class ScammerRegisterRequest(BaseModel):
    phone_number: str
    last_message: str = ""
    risk_level: str = "HIGH"
    is_active: bool = True


class ScammerHistoryItem(BaseModel):
    role: str
    content: str
    timestamp: datetime


class ScammerHistoryResponse(BaseModel):
    phone_number: str
    ai_enabled: bool
    history: list[ScammerHistoryItem]


class ActivateAIRequest(BaseModel):
    enabled: bool = True

File: android/app/src/main/java/com/scamshield/app/data/remote/ScammerModels.kt
package com.scamshield.app.data.remote

data class ScammerRegisterRequestDto(
    val phone_number: String,
    val last_message: String,
    val risk_level: String = "HIGH",
    val is_active: Boolean = true
)

data class ActivateAiRequestDto(
    val enabled: Boolean
)

data class ScammerHistoryResponseDto(
    val phone_number: String,
    val ai_enabled: Boolean,
    val history: List<ScammerHistoryItemDto>
)

data class ScammerHistoryItemDto(
    val role: String,
    val content: String,
    val timestamp: String
)
